test/functional/specs/Personalization/C22098199.js

```diff
@@ -0,0 +1,143 @@
+});
+
+const containerId = "target-container";
+const testPageBody = `
+<div id="container">
+  <h1 id="page-header">Hello World!</h1>
+  <div id="${containerId}"></div>
+</div>
+`;
+
+const createPropositions = ({
+  propositionId = uuid(),
+  viewName = "test",
+  activityId = uuid(),
+  items = [],
+}) => {
+  return [
+    {
+      id: propositionId,
+      scope: `web://aepdemo.com/#${viewName}`,
+      scopeDetails: {
+        decisionProvider: ADOBE_JOURNEY_OPTIMIZER,
+        correlationID: "6dae465b-9553-4fc6-b7d4-6c9979c88f21-0",
+        characteristics: {
+          eventToken:
+            "eyJtZXNzYWdlRXhlY3V0aW9uIjp7Im1lc3NhZ2VFeGVjdXRpb25JRCI6IlVFOkluYm91bmQiLCJtZXNzYWdlSUQiOiJmMzgxZWJhYS1kNDIyLTQxNzQtOWUzNS0yMTY3NDYwMjk5MTAiLCJtZXNzYWdlUHVibGljYXRpb25JRCI6IjZkYWU0NjViLTk1NTMtNGZjNi1iN2Q0LTZjOTk3OWM4OGYyMSIsIm1lc3NhZ2VUeXBlIjoibWFya2V0aW5nIiwiY2FtcGFpZ25JRCI6ImVhZDg5MWE0LTNjYWUtNGE1ZC05MGEzLTFkZTc0MzkwYjNkMyIsImNhbXBhaWduVmVyc2lvbklEIjoiZDhiYzk5YmMtZGRhZC00Y2MyLThlYjItYTJlMGUzY2FmNzg0IiwiY2FtcGFpZ25BY3Rpb25JRCI6IjQzNmZmM2NkLTZkZWItNDczNi04NDc1LTA3NDhhYzc4MTlkOCJ9LCJtZXNzYWdlUHJvZmlsZSI6eyJtZXNzYWdlUHJvZmlsZUlEIjoiMDg5NGYwNmYtOTkyNi00YTc2LTk4OTktYThmZjc3NWZmNTA4IiwiY2hhbm5lbCI6eyJfaWQiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbHMvd2ViIiwiX3R5cGUiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbC10eXBlcy93ZWIifX19",
+          viewName,
+          scopeType: "view",
+        },
+        activity: {
+          id: activityId,
+        },
+      },
+      items: [...items],
+    },
+  ];
+};
+
+/**
+ * @param {string} action
+ */
+const createTest = (action) => async () => {
+  const config = compose(orgMainConfigMain, debugEnabled);
+  const propositionId = uuid();
+  const itemId = uuid();
+
+  const contentId = "prepended-content";
+  const propositions = createPropositions({
+    propositionId,
+    items: [
+      {
+        id: itemId,
+        schema: "https://ns.adobe.com/personalization/dom-action",
+        data: {
+          type: action,
+          content: `<div id="${contentId}">This is content—${action}</div>`,
+          selector: `#${containerId}`,
+        },
+      },
+    ],
+  });
+
+  await addHtmlToBody(testPageBody, true);
+
+  const alloy = createAlloyProxy();
+  await alloy.configure(config);
+
+  // First application
+  await alloy.applyPropositions({
+    propositions,
+  });
+
+  const container = Selector(`#${containerId}`).addCustomDOMProperties({
+    renderedIds: (el) => el.dataset.adobePropositionIds,
+  });
+  const renderedContent = Selector(`#${contentId}`);
+  // Verify the content was prepended once
+  await t.expect(renderedContent.count).eql(1);
+  await t.expect(container.renderedIds).ok();
+  await t.expect(container.renderedIds).contains(itemId);
+
+  // Second application - should not append again
+
+  await alloy.applyPropositions({
+    propositions,
+  });
+
+  // Verify the content was not appended again
+  await t.expect(renderedContent.count).eql(1);
+};
```

Comment:
Can you add sendEvent calls with "includeRenderedPropositions: true" after the applyProposition calls to check that the display notification are still sent even when not rendered? Also can you add a validation on the return value of applyPropositions that the proposition was marked rendered?